ktor-client/ktor-client-java/jvm/src/io/ktor/client/engine/java/JavaHttpWebSocket.kt

```diff
@@ -183,7 +183,7 @@ internal class JavaHttpWebSocket(
     }.asCompletableFuture()
 
     override fun onError(webSocket: WebSocket, error: Throwable) {
-        val cause = WebSocketException("${error.message}")
+        val cause = WebSocketException("$error.message}", error)
```

Comment:
`"$error.message}"` -> `"${error.message}"` or maybe add WebSocket-specific message? Something generic like `WebSocket failed`

Reply:
Oh... thanks for spotting this 🤦 I had actually put `error.message ?: "generic message here"` and then I reverted to keep the PR focused, but reverting manually is a bad idea as we can see!

I like the `web socket failed` generic message. Do you usually spell it in one word in Ktor's material?

Reply:
I think we would benefit from adding the cause's message to that new exception message, so if some code just prints the message of Ktor's exception (and not the cause) then the error still contains the information.

And since the generic message doesn't bring any useful information on top of the exception name, we could actually only use it in case the cause's message is absent. Something like:

```kotlin
val cause = WebSocketException(error.message ?: "web socket failed", error)
```

Reply:
Yes, sounds good.